### ISSUE TITLE
vita2d_set_region_clip: use sceGxmColorSurfaceSetClip instead of sceGxmSetRegionClip to have pixel granularity?

### ISSUE THREAD

**Issue by Lupo511** [xerpi/vita2dlib#39]
Why does vita2d_set_region_clip use `sceGxmSetRegionClip` instead of `sceGxmColorSurfaceSetClip`?
Currently `sceGxmSetRegionClip` allignes the region to tiles (unless I'm wrong), while `sceGxmColorSurfaceSetClip` should actually set a per-pixel region.

**Comment by xerpi** [comment]
I didn't know *sceGxmColorSurfaceSetClip* existed, feel free to send a PR.

**Comment by Lupo511** [comment]
I found out that `sceGxmColorSurfaceSetClip` only works before you begin the scene, so it's not as useful as I though. Do you want me to still add it as a separate function, as it might have some use?

By the way what I needed was to change the region clip between different draw calls, do you think that modifying the fragment shaders to implement a custom clipping system (discard pixels outside of a region) would be viable?

**Comment by xerpi** [comment]
We could cache the set_clip call values and then call sceGxmColorSurfaceSetClip before the beginScene().
Yeah discarding on the fragment shaders would also work, but it's too much work imho.

**Comment by Lupo511** [comment]
Caching the set_clip calls still wouldn't allow to change the region clip during a scene tough.

**Comment by xerpi** [comment]
Yeah.. Another option would be using a stencil buffer, but I have no idea how it works.

**Comment by Lupo511** [comment]
That is actually a nice idea and seems t be working. I made a pull request (#40) that adds it, see what you think!
